drivers/pds/lib/utils.go

```diff
@@ -245,19 +280,88 @@ func SetupPDSTest(ControlPlaneURL, ClusterType, TargetClusterName, AccountName s
 	return tenantID, dnsZone, projectID, serviceType, deploymentTargetID, err
 }
 
-// GetClusterID retruns the cluster id for given targetClusterName
-func GetClusterID(projectID string, targetClusterName string) (string, error) {
-	deploymentTargets, err := components.DeploymentTarget.ListDeploymentTargetsBelongsToProject(projectID)
+// CheckNamespace checks if the namespace is available in the cluster and pds is enabled on it
+func CheckNamespace(namespace string) (bool, error) {
+	ns, err = k8sCore.GetNamespace(namespace)
+	isavailable = false
 	if err != nil {
-		logrus.Errorf("An Error Occured while listing deployment targets %v", err)
-		return "", err
+		logrus.Warnf("Namespace not found %v", err)
+		if strings.Contains(err.Error(), "not found") {
+			nsName := &corev1.Namespace{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:   namespace,
+					Labels: map[string]string{pxLabel: "true"},
+				},
+			}
+			logrus.Infof("Creating namespace %v", namespace)
+			ns, err = k8sCore.CreateNamespace(nsName)
+			if err != nil {
+				logrus.Errorf("Error while creating namespace %v", err)
+				return false, err
+			}
+			isavailable = true
+		}
+		if !isavailable {
+			return false, err
+		}
 	}
-	for index := range deploymentTargets {
-		if deploymentTargets[index].GetName() == targetClusterName {
-			return deploymentTargets[index].GetClusterId(), nil
+	isavailable = false
+	for key, value := range ns.Labels {
+		logrus.Infof("key: %v values: %v", key, value)
+		if key == pxLabel && value == "true" {
+			isavailable = true
+			break
+		}
+	}
+	if !isavailable {
+		return false, nil
+	}
+	return true, nil
+}
+
+// ReadParams reads the params from given or default json
+func ReadParams(filename string) (*Parameter, error) {
+	var jsonPara Parameter
+
+	if filename == "" {
+		filename, err = filepath.Abs(defaultParams)
+		logrus.Infof("filename %v", filename)
+		if err != nil {
+			logrus.Errorf("FilePath err %v", err)
+			return nil, err
+		}
```

Comment:
Can we please not have this, if you want to have verbose error, just return verbose error
```
		if err != nil {
			return nil, fmt.Errorf("FilePath err %v", err)
		}
```
There are too many places to mention this, can you please fix it everywhere you have this